### PR TITLE
fix(pipeline): optimization of pipeline k8s executor initialization

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
optimization of pipeline k8s executor initialization

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda-org.erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJhc3NpZ25lZSI6WyIxMDAxMjA1Il19&id=286324&iterationID=1115&pId=0&type=TASK)

#### Specified Reviewers:

/assign @your-reviewer

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |Fix the bug that optimization of pipeline k8s executor initialization              |
| 🇨🇳 中文    |     修复pipeline启动时多集群耗时慢的问题         |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/manager.go [hunk 1]
```diff
@@ -75,7 +75,9 @@ func (m *Manager) Initialize() error {
 
 	logrus.Infof("pipeline scheduler task executor Inititalize ...")
 
-	if err := m.batchUpdateExecutors(); err != nil {
+	// initialization only need to set up all clusters
```
Comment (sfwn) [inline modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/manager.go:78]: It's two sentences.
add `.` at the end of each sentence.

File: modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/manager.go [hunk 2]
```diff
@@ -248,6 +254,19 @@ func (m *Manager) batchUpdateExecutors() error {
 	return nil
 }
 
+func (m *Manager) batchSetupClusters() error {
+	clusters, err := clusterinfo.ListAllClusters()
+	if err != nil {
+		return err
+	}
+	for i := range clusters {
```
Comment (sfwn) [inline modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/manager.go:262]: lock before for-loop is enough.

File: modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/manager.go [hunk 3]
```diff
@@ -75,7 +75,9 @@ func (m *Manager) Initialize() error {
 
 	logrus.Infof("pipeline scheduler task executor Inititalize ...")
 
-	if err := m.batchUpdateExecutors(); err != nil {
+	// initialization only need to set up all clusters
+	// executor could be created when pipeline execution or async created
+	if err := m.batchSetupClusters(); err != nil {
```
Comment (sfwn) [inline modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/manager.go:80]: can we setup clusters when needed?

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/manager.go (hunk 2 above)
Comment (chengjoey) [reply modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/manager.go]: adjusted

File: modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/manager.go (hunk 3 above)
Comment (chengjoey) [reply modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/manager.go]: adjusted, setup cluster like executor

**Review by sfwn — APPROVED** [review]

**Pull request merged**